uaclient/clouds/tests/test_aws.py

```diff
@@ -17,6 +20,44 @@ def test_identity_doc_from_aws_url_pkcs7(self, readurl):
         url = "http://169.254.169.254/latest/dynamic/instance-identity/pkcs7"
         assert [mock.call(url)] == readurl.call_args_list
 
+    @pytest.mark.parametrize("caplog_text", [logging.DEBUG], indirect=True)
+    @pytest.mark.parametrize(
+        "fail_count,exception", ((3, None), (4, HTTPError))
+    )
+    @mock.patch(M_PATH + "util.time.sleep")
+    @mock.patch(M_PATH + "util.readurl")
+    def test_retry_backoff_on_failed_identity_doc(
+        self, readurl, sleep, fail_count, exception, caplog_text
+    ):
+        """Retry backoff is attempted before failing to get AWS.identity_doc"""
+
+        def fake_someurlerrors(url):
+            if readurl.call_count <= fail_count:
+                raise HTTPError(
+                    None, 400 + readurl.call_count, None, None, BytesIO()
+                )
+            return "pkcs7WOOT!==", {"header": "stuff"}
+
+        readurl.side_effect = fake_someurlerrors
+        instance = UAPremiumAWSInstance()
+        if exception:
+            with pytest.raises(HTTPError) as excinfo:
```

Comment:
A nit, but if we're going to hard-code `HTTPError` here (which is fine!), we can make `exception` just a `bool`. (Alternatively, we could do `pytest.raises(exception)` here, but I think that's less obvious for no real gain.)